static/js/chart/draw.ts

```diff
@@ -427,10 +429,174 @@ function drawLineChart(
+    maxV = Math.max(maxV, Math.max(...dataGroupsDict[geoId].map((dataGroup) => dataGroup.max())));
+
+  }
+
+  let svg = d3
+      .select("#" + id)
+      .append("svg")
+      .attr("width", width)
+      .attr("height", height);
+
+  let xScale = d3
+      .scaleTime()
+      .domain(d3.extent(dataGroups[0].value, (d) => new Date(d.label).getTime()))
+      .range([MARGIN.left, width - MARGIN.right - MARGIN.legendWidth]);
+
+  let yScale = d3
+      .scaleLinear()
+      .domain([minV, maxV])
+      .range([height - MARGIN.bottom, MARGIN.top])
+      .nice(NUM_Y_TICKS);
+
+  addXAxis(svg, height, xScale);
+  addYAxis(svg, width, yScale, unit);
+
+
+  let dashIndex = 0;
+  for (let geoId in dataGroupsDict) {
+    dataGroups = dataGroupsDict[geoId];
+    dashes.push(DASHES[dashIndex]);
+    for (let i = 0; i < dataGroups.length; i++) {
+      let dataGroup = dataGroups[i];
+      let dataset = dataGroup.value.map(function (dp) {
+        return [new Date(dp.label).getTime(), dp.value];
+      });
+
+      if (dashIndex == 0) {
+        colors.push(colorFn(dataGroupsDict[geoId][i].label));
+      }
+
+      let color = colors[i];
+
+      let line = d3
+          .line()
+          .x((d) => xScale(d[0]))
+          .y((d) => yScale(d[1]));
+
+      svg
+          .append("path")
+          .datum(dataset)
+          .attr("class", "line")
+          .style("stroke", color)
+          .attr("d", line)
+          .attr("stroke-width", "2")
+          .attr("stroke-dasharray", dashes[dashIndex]);
+    }
+
+    dashIndex++;
+  }
+
+  svg
+      .append("g")
+      .attr("id", "inChartLegend")
+      .attr("transform", `translate(${width - MARGIN.legendWidth}, 50)`);
+
+  buildLegend("inChartLegend", colors, dashes, dataGroupsDict);
+
+  // return colors here used to add menu below the chart.
+  return colors;
+}
+
+
+/**
+ * Generate in-chart legend.
+ *
+ * @param id
+ * @param colors
+ * @param dashes
+ * @param dataGroupsDict
+ */
+function buildLegend(
+    id: string,
+    colors: String [],
+    dashes: String [],
+    dataGroupsDict: {[geo: string]: DataGroup[];},
+) {
+  let legend = d3
+      .select("#" + id);
+
+  let index = 0;
+  for (let geo in dataGroupsDict) {
+    let legendClass = legend
+        .append("g")
+        .attr("transform", `translate(0, ${20 * index})`);
+
+    if (dashes.length == 1) {
+      legendClass
+          .append("line")
+          .attr("stroke-width", 2)
+          .attr("stroke-dasharray", "")
+          .attr("x1", "0")
+          .attr("y1", `${20 * index - 5}`)
+          .attr("x2", "30")
+          .attr("y2", `${20 * index - 5}`)
+          .attr("stroke", `${colors[index]}`);
+    } else {
+      legendClass
+          .append("line")
+          .attr("stroke-width", 2)
+          .attr("stroke-dasharray", "")
+          .attr("x1", "0")
+          .attr("y1", `${20 * index - 5}`)
+          .attr("x2", "30")
+          .attr("y2", `${20 * index - 5}`)
+          .attr("stroke", "#930000")
```

Comment:
Why is this the default stroke color if there aren't any dashes? What else does the length of the dashes array imply, and can we be more explicit about it?